### ISSUE TITLE
Dependency version updates

### ISSUE THREAD

**Issue by chandakakshat** [graphql-java/graphql-java#3263]
Even though this is about a Guava vulnerability, because graphql-java shades the Guava library, vulnerabilities-scanning software is flagging graphql-java as also being vulnerable.

Graphql-java seems to be using few modules of guava the fix for vulnerability in [release 20.4](https://github.com/graphql-java/graphql-java/releases/tag/v20.4) the 20.4 version uses 32.0.0 version of guava which still contains the vulnerablity[ CVE-2023-2976](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-2976)

I am creating this ticket to ask if you could release a patch version including this update and to use [guava versions](https://mvnrepository.com/artifact/com.google.guava/guava) **>=32.0.1** to remove the vulnerablity

The Guava vulnerability reported  has been fixed in Guava version >=32.0.1.

**Comment by bclozel** [comment]
CVE-2023-2976 is fixed in 32.0.0, it's just that this [version breaks a feature in windows](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-2976):

> Even though the security vulnerability is fixed in version 32.0.0, we recommend using version 32.0.1 as version 32.0.0 breaks some functionality under Windows. 

As discussed in [the official announcement](https://github.com/graphql-java/graphql-java/discussions/3246), this project doesn't use the Guava offending classes so the upgrade was mostly done to make security tools happy. The advisory you're pointing to (with your last link) is not related to the case discussed here. Thoughts?

**Comment by chandakakshat** [comment]
> [CVE-2023-2976](https://github.com/advisories/GHSA-7g45-4rm6-3mm3) is fixed in 32.0.0, it's just that this [version breaks a feature in windows](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-2976):
> 
> > Even though the security vulnerability is fixed in version 32.0.0, we recommend using version 32.0.1 as version 32.0.0 breaks some functionality under Windows.
> 
> As discussed in [the official announcement](https://github.com/graphql-java/graphql-java/discussions/3246), this project doesn't use the Guava offending classes so the upgrade was mostly done to make security tools happy. The advisory you're pointing to (with your last link) is not related to the case discussed here. Thoughts?

Thanks that clears everything, But the security tools do not seem to be happy with 32.0.0 and suggests the versions >=32.0.1 to address the vulnerability :)

**Comment by bclozel** [comment]
@chandakakshat that's really feedback that should be sent to the security tool vendor . The graphql-java team might upgrade in the future anyway, but this has zero added value since they were not vulnerable in the first place, and here this is a case of an invalid alert by the security tool. Can you create an issue with the security tool vendor and share the link here?

**Comment by dondonz** [comment]
Thanks @bclozel !

In master we're now using Guava 32.1.1 and this will be included in the next release happening very soon (later this month).

Let me know @chandakakshat how you go with the security tool vendor. We did backport the CVE fix already, however if more people in the community want the patch with >=32.0.1, I am open to the idea if more people ask

**Comment by chandakakshat** [comment]
> Thanks @bclozel !
> 
> In master we're now using Guava 32.1.1 and this will be included in the next release happening very soon (later this month).
> 
> Let me know @chandakakshat how you go with the security tool vendor. We did backport the CVE fix already, however if more people in the community want the patch with >=32.0.1, I am open to the idea if more people ask

Sure, Thanks @bclozel @dondonz for the help 
I've raised an issue with the vendor, lets see how it goes -- awaiting for a reply

**Comment by chandakakshat** [comment]
Hii @dondonz will we be getting a patch version for 18.x versions ?

**Comment by brfrn169** [comment]
The Trivy vulnerability scanner also recommends using a version with >= `32.0.1`, as follows:

```
Java (jar)
==========
Total: 1 (HIGH: 1, CRITICAL: 0)

┌────────────────────────────────────────────────┬───────────────┬──────────┬───────────────────┬────────────────┬───────────────────────────────────────────┐
│                    Library                     │ Vulnerability │ Severity │ Installed Version │ Fixed Version  │                   Title                   │
├────────────────────────────────────────────────┼───────────────┼──────────┼───────────────────┼────────────────┼───────────────────────────────────────────┤
│ com.google.guava:guava (graphql-java-20.4.jar) │ CVE-2023-2976 │ HIGH     │ 32.0.0-jre        │ 32.0.1-android │ insecure temporary directory creation     │
│                                                │               │          │                   │                │ https://avd.aquasec.com/nvd/cve-2023-2976 │
└────────────────────────────────────────────────┴───────────────┴──────────┴───────────────────┴────────────────┴───────────────────────────────────────────┘
```

Are you planning to release a patch version for `20.4` using Guava version >= `32.0.1`?

**Comment by bclozel** [comment]
@brfrn169 did you report the metadata error to Trivy? I think suggesting `32.0.1-android` as a fix version is wrong for two reasons:

* [32.0.1 doesn't fix the CVE](https://github.com/graphql-java/graphql-java/issues/3263#issuecomment-1619727883)
* this changes the variant from `-jre` to `-android`

I think it's a good idea to help vendors improve the metadata instead of putting more pressure on maintainers for the wrong reasons (graphql-java is not vulnerable in the first place).

**Comment by brfrn169** [comment]
@bclozel Thank you for your advice. Filed in https://github.com/aquasecurity/trivy/discussions/4839.

**Comment by dondonz** [comment]
Hey folks thanks for reporting. Although graphql-java is not vulnerable, as a fellow dev I feel your pain of the security scanners. I'll release new versions with updated POMs soon

**Comment by brfrn169** [comment]
FYI: Trivy has handled this issue:
https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/issues/236

Thanks.

**Comment by bclozel** [comment]
@brfrn169 thanks for filing the issue. You've improved the situation for lots of developers.

**Comment by dondonz** [comment]
Thanks for filing the issue @brfrn169 and glad to see the security team updated their advice. @chandakakshat were you also able to update the security advice for your scanner tool?

**Comment by chandakakshat** [comment]
> Thanks for filing the issue @brfrn169 and glad to see the security team updated their advice. @chandakakshat were you also able to update the security advice for your scanner tool?

My security scanner apparently does not suggest -android for -jre variant

**Comment by chandakakshat** [comment]
Hi @dondonz, Can you provide a patch for 18.x version also?

**Comment by dondonz** [comment]
@chandakakshat yes let's do it, I'd rather keep the scanner happy. I understand how hard it can be to change the scanner rules

**Comment by dondonz** [comment]
Thanks @chandakakshat for your PR, I have released 18.7 and it should land in Maven shortly.

Release notes: https://github.com/graphql-java/graphql-java/releases/tag/v18.7

To anyone else reading, we are going to release the v19 and v20 versions as well, we wanted to bundle in an extra 2 bugfixes for v19 and v20.

**Comment by chandakakshat** [comment]
> Thanks @chandakakshat for your PR, I have released 18.7 and it should land in Maven shortly.
> 
> Release notes: https://github.com/graphql-java/graphql-java/releases/tag/v18.7
> 
> To anyone else reading, we are going to release the v19 and v20 versions as well, we wanted to bundle in an extra 2 bugfixes for v19 and v20.

Thank you @dondonz @bclozel, for your support!

**Comment by dondonz** [comment]
Today we released 19.7 and 20.5. Both releases contain Guava 32.1.1 which should be accepted as patched by all security scanners. The previously released 21.0 already contained Guava 32.1.1.

Thanks everyone for raising the issue!

Release notes
https://github.com/graphql-java/graphql-java/releases/tag/v19.7
https://github.com/graphql-java/graphql-java/releases/tag/v20.5
